src/rpc/blockchain.cpp

```diff
@@ -985,6 +987,200 @@ static RPCHelpMan gettxoutsetinfo()
+
+                const CAmount subsidy = cumulative ? stats.total_subsidy : stats.total_subsidy - prev_stats.total_subsidy;
+                file << strprintf("%s,", FormatMoney(subsidy));
+
+                CHECK_NONFATAL(stats.total_amount.has_value());
+                const CAmount amount = cumulative ? stats.total_amount.value() : stats.total_amount.value() - prev_stats.total_amount.value();
+                file << strprintf("%s,", FormatMoney(amount));
+
+                const CAmount unspendables_unclaimed_rewards = cumulative ? stats.total_unspendables_unclaimed_rewards : stats.total_unspendables_unclaimed_rewards - prev_stats.total_unspendables_unclaimed_rewards;
+                file << strprintf("%s,", FormatMoney(unspendables_unclaimed_rewards));
+
+                const CAmount prevout_spent_amount = cumulative ? stats.total_prevout_spent_amount : stats.total_prevout_spent_amount - prev_stats.total_prevout_spent_amount;
+                file << strprintf("%s,", FormatMoney(prevout_spent_amount));
+
+                const CAmount new_outputs_ex_coinbase_amount = cumulative ? stats.total_new_outputs_ex_coinbase_amount : stats.total_new_outputs_ex_coinbase_amount - prev_stats.total_new_outputs_ex_coinbase_amount;
+                file << strprintf("%s,", FormatMoney(new_outputs_ex_coinbase_amount));
+
+                const CAmount unspendables_scripts = cumulative ? stats.total_unspendables_scripts : stats.total_unspendables_scripts - prev_stats.total_unspendables_scripts;
+                file << strprintf("%s,", FormatMoney(unspendables_scripts));
+
+                const CAmount unspendable_amount = cumulative ? stats.total_unspendable_amount : stats.total_unspendable_amount - prev_stats.total_unspendable_amount;
+                file << strprintf("%s,", FormatMoney(unspendable_amount));
+
+                const CAmount unspendables_genesis_block = cumulative ? stats.total_unspendables_genesis_block : stats.total_unspendables_genesis_block - prev_stats.total_unspendables_genesis_block;
+                file << strprintf("%s,", FormatMoney(unspendables_genesis_block));
+
+                const CAmount unspendables_bip30 = cumulative ? stats.total_unspendables_bip30 : stats.total_unspendables_bip30 - prev_stats.total_unspendables_bip30;
+                file << strprintf("%s,", FormatMoney(unspendables_bip30));
+
+                const arith_uint256 chain_work = cumulative ? pindex->nChainWork : pindex->nChainWork - (pindex->pprev ? pindex->pprev->nChainWork : 0);
+                file << strprintf("%s,", chain_work.GetHex());
+
+                file << strprintf("%d,", stats.hashSerialized.GetHex());
+
+                const uint64_t bogo_size = cumulative ? stats.nBogoSize : stats.nBogoSize - prev_stats.nBogoSize;
+                file << strprintf("%d\n", static_cast<int64_t>(bogo_size));
+
+                prev_stats = stats;
+
+                if (pindex->nHeight >= active_chainstate.m_chain.Tip()->nHeight) break;
```

Comment:
Is it possible that the index has not synced up to `active_chainstate.m_chain.Tip()` if the tip moved during the dump? I don't see any lock for that. Could be fixed with this?
```suggestion
                if (pindex->nHeight >= g_coin_stats_index->GetSummary().best_block_height) break;
```

Reply:
Yeah, I agree, there is probably a very small chance of this happening and it is avoided with this change.